### PR TITLE
unified layer api

### PR DESCRIPTION
**Pull request by wilsaj**
add a level of abstraction for layers
- provides a single API for the Map component, so stuff that is specific to a layer type can stay separate
- gives us a place to put shared code to manage things like whether or not a layer is ready to be added to the map, since layers require a series of API calls to determine things like actual tile URLs

### CODE REVIEW

**Comment by wilsaj** [comment]
base layers could also use this, but not as important since there is no ready state to manage

File: web-viewer/src/components/Map.jsx [hunk 1]
```diff
@@ -14,7 +13,6 @@ const Map = React.createClass({
       layers: PropTypes.arrayOf(CustomPropTypes.baseLayer),
       active: PropTypes.string
     }),
-    activeWeatherLayerId: PropTypes.string
   },
   getInitialState() {
     this.layerCache = new LayerCache()
```
Comment (jseppi) [inline web-viewer/src/components/Map.jsx]: I think semantically that this should go in `componentDidMount` or `initializeLayerCache`

File: web-viewer/src/util/aeris.es [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    if (this.weatherLayerTimeout) {
+      clearTimeout(this.weatherLayerTimeout)
+      this.weatherLayerTimeout = null
+    }
+
+    this.layers.forEach((layer) => {
+      if (map.hasLayer(layer)) {
+        map.removeLayer(layer)
+      }
+    })
```
Comment (jseppi) [inline web-viewer/src/util/aeris.es]: Perhaps also set `this.layers = []`

**Comment by jseppi** [comment]
:+1: great idea - minor comments above

**Pull request merged**